server/src/testFixtures/java/io/crate/testing/TestingHelpers.java

```diff
@@ -405,4 +411,23 @@ public static <T> List<T> getRandomsOfType(int minLength, int maxLength, DataTyp
         }
         return values;
     }
+
+    public static Map<String, Object> toMapping(BoundCreateTable boundCreateTable) {
+        LinkedHashMap<ColumnIdent, Reference> references = new LinkedHashMap<>();
+        IntArrayList pKeysIndices = new IntArrayList();
+        boundCreateTable.analyzedTableElements().collectReferences(boundCreateTable.tableIdent(), references, pKeysIndices, true);
+        var policy = (String) boundCreateTable.tableParameter().mappings().get(ColumnPolicies.ES_MAPPING_NAME);
+        var tableColumnPolicy = policy != null ? ColumnPolicies.decodeMappingValue(policy) : ColumnPolicy.STRICT;
+
+        return createMapping(
+            new ArrayList<>(references.values()),
+            pKeysIndices,
+            boundCreateTable.analyzedTableElements().getCheckConstraints(),
+            boundCreateTable.analyzedTableElements().indicesMap(),
+            boundCreateTable.partitionedBy(),
+            tableColumnPolicy,
+            boundCreateTable.routingColumn()
+        );
+
+    }
```

Comment:
Looks like similar code is used at the `TableCreator`. Maybe worth moving it to `MappingUtil` with a `@deprecated` annotation?

Reply:
If I mark it as `@deprecated`, it has to be removed. I think `TestingHelpers.toMapping` cannot be removed as it's the only way to get mapping in analysis stage which is needed for unit tests setup, like `SQLExecutor.addTable`.

I was going to maybe reuse TestingHelper.toMapping in TableCreator but it looked weird to use "Test" class there. I think it's fine to keep TableCreator version since it can be removed in 5.5.

Or you meant moving it to `MappingUtil` and then get back to `TestingHelpers` in 5.5 once we remove `TableCreator` BWC branch and it will become test-framework only code?

Reply:
> If I mark it as `@deprecated`, it has to be removed. I think `TestingHelpers.toMapping` cannot be removed as it's the only way to get mapping in analysis stage which is needed for unit tests setup, like `SQLExecutor.addTable`.

Ah right, ok lets leave it like that.